plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+- Authenticating
+  - The fallback auth UI
+- Performing queries
+  - Recommendations for caching data
+  - How the Coder plugin caches data
+  - The `useCoderQuery` custom hook
+  - Cautions against other common UI caching strategies
+- Performing mutations
+
+### Before you begin
+
+This guide assumes that you have already added the `CoderProvider` component to your Backstage deployment. If you have not, [please see the main README](../../README.md#setup) for instructions on getting that set up.
+
+## Quick-start
+
+### The main hooks to use when working with the SDK
+
+There are about six main hooks that you want to consider when using the Coder SDK.These can be broken down into two main categories:
+
+#### Primitive hooks
+
+- `useCoderSdk` - Gives you the Coder SDK instance
+- `useCoderAuth` - Exposes properties and methods for examining/updating Coder auth state
+- `useQuery` (via Tanstack Query library) - Declarative querying
+- `useMutation` (via Tanstack Query library) - Imperative mutations
+- `useQueryClient` (via Tanstack Query library) - Lets you coordinate queries and mutations within the same query cache
+
+#### Convenience hooks
+
+- `useCoderQuery` (wrapper over `useQuery`)
+
+## Accessing the SDK from your own custom React components
+
+There are two main ways of accessing the Coder SDK:
+
+- `useCoderSdk`
+- `useCoderQuery`
+
+### Accessing the SDK through `useCoderSdk`
+
+`useCoderSdk` is a lower-level "primitive" for accessing the Coder SDK. The SDK exposes a mix of different REST API methods for interacting with your Coder deployment's resources.
+
+Calling the hook will give you an object with all available API methods. As these methods are all async, **none** of them are suitable for use in render logic. They must be called from within effects or event handlers.
+
+```tsx
+// Illustrative example - these patterns are a very bad idea in production!
+function ExampleComponent() {
+  // The workspace type is exported via the plugin
+  const [workspaces, setWorkspaces] = useState<readonly Workspace[]>([]);
+  const sdk = useCoderSdk();
+
+  // The SDK can be called from any effect
+  useEffect(() => {
+    const getInitialWorkspaces = async () => {
+      const workspacesResponse = await sdk.getWorkspaces({ q: 'owner:me' });
+      const workspaces = workspacesResponse.workspaces;
+      setWorkspaces(workspaces);
+    };
+
+    void getInitialWorkspaces();
+
+    // The SDK maintains a stable memory reference; there is no harm in
+    // including it as part of your dependency arrays. In this case, the
+    // dependency array may as well be empty.
+  }, [sdk]);
+
+  return (
+    <>
+      <button
+        // The SDK can also be called from any event handler
+        onClick={async () => {
+          const newWorkspace = await sdk.createWorkspace(
+            'organizationId',
+            'userId',
+            {
+              // Properties for making new workspace go here
+            },
```

Comment:
Can we fill this with properties so it works out of the box? Or explain to the user where they can find the properties they need to fill? As a non-Typescript expert, I may get stuck here.